components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLModelHandler.java

```diff
@@ -368,6 +369,44 @@ public MLModel retrieveModel(long modelId) throws MLModelBuilderException {
         }
     }
 
+    /**
+     * Publish a ML model to registry.
+     *
+     * @param tenantId    Unique ID of the tenant.
+     * @param userName    Username of the user.
+     * @param modelId     Unique ID of the built ML model.
+     * @throws MLModelBuilderException
+     */
+    public void publishModel(int tenantId, String userName, long modelId) throws MLModelBuilderException {
+        InputStream in = null;
+        try {
+            // read model
+            MLStorage storage = databaseService.getModelStorage(modelId);
+            String storageType = storage.getType();
+            String storageLocation = storage.getLocation();
+            MLIOFactory ioFactory = new MLIOFactory(mlProperties);
+            MLInputAdapter inputAdapter = ioFactory.getInputAdapter(storageType + MLConstants.IN_SUFFIX);
+            in = inputAdapter.read(new URI(storageLocation));
+            // create registry path
+            MLCoreServiceValueHolder valueHolder = MLCoreServiceValueHolder.getInstance();
+            String modelName = databaseService.getModel(tenantId, userName, modelId).getName();
+            String registryPath = File.separator + valueHolder.getModelRegistryLocation() + File.separator + modelName;
+            // publish to registry
+            RegistryOutputAdapter registryOutputAdapter = new RegistryOutputAdapter();
+            registryOutputAdapter.write(registryPath, in);
+
+        } catch (Exception e) {
+            throw new MLModelBuilderException("Failed to publish the model [id] " + modelId, e);
```

Comment:
Introduce a new exception and throw it here, since this is not related to model building but publishing.